File: frontend/src/types/inspection.ts
export type InspectionVerdict = 'PASS' | 'FAIL' | 'UNCLEAR'

export type InspectionCriterion = {
  name: string
  verdict: InspectionVerdict
  note: string
}

export type InspectionFindings = {
  criteria: InspectionCriterion[]
  summary: string | null
}

export type SpecInspection = {
  id: number
  spec_id: number
  model_id: number
  selected_model_id: number | null
  selected_model_name: string | null
  selected_prompt_id: number | null
  selected_prompt_name: string | null
  findings: InspectionFindings
  passes: number
  created_at: string
}

File: backend/app/schemas/inspection.py
"""Inspection API schemas."""
from pydantic import BaseModel, Field


class InspectRequest(BaseModel):
    """Request body for running a Spec inspection."""

    model_id: int | None = None


class InspectionCriterion(BaseModel):
    """One parsed inspection criterion."""

    name: str
    verdict: str
    note: str


class InspectionFindings(BaseModel):
    """Parsed inspection findings."""

    criteria: list[InspectionCriterion]
    summary: str | None = None


class SpecInspectionOut(BaseModel):
    """Persisted inspection response."""

    id: int
    spec_id: int
    model_id: int
    selected_model_id: int | None = None
    selected_model_name: str | None = None
    findings: InspectionFindings
    summary: str | None = None
    passes: int = Field(ge=1)
    created_at: str
